jax/_src/numpy/linalg.py

```diff
@@ -238,11 +238,32 @@ def _cofactor_solve(a, b):
   return partial_det[..., -1], x
 
 
+def _det_3x3(a):
+  return (a[..., 0, 0] * a[..., 1, 1] * a[..., 2, 2] +
+          a[..., 0, 1] * a[..., 1, 2] * a[..., 2, 0] +
+          a[..., 0, 2] * a[..., 1, 0] * a[..., 2, 1] -
+          a[..., 0, 2] * a[..., 1, 1] * a[..., 2, 0] -
+          a[..., 0, 0] * a[..., 1, 2] * a[..., 2, 1] -
+          a[..., 0, 1] * a[..., 1, 0] * a[..., 2, 2])
+
+@jit
+def _det(a):
```

Comment:
I think you can just stick a `@jit` around `det`, no need for another `_det` function.

(Sometimes we use that pattern when we want to handle static arguments specially, but `det` doesn't have any.)

Reply:
kk--was about to ask about this actually :D

Reply:
BTW the jit should probably be innermost (i.e., closest to the `def`)

Reply:
++

Reply:
done, I think!